# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerOpPromise.h
# Commit: cf972fe0d961
# Full Hash: cf972fe0d96164788f998f04008930474526e400
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-02-26 21:36:36
# Regressor Bug: 1754365
# File Overlap Count: 1
# Description:
#   Bug 1754365 - Create IPC for transferring NavigationPreload's PerformanceTimingData. r=dom-worker-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138251
# ==============================================================================

diff -r f8afd25bf41e -r cf972fe0d961 dom/serviceworkers/ServiceWorkerOpPromise.h
--- a/dom/serviceworkers/ServiceWorkerOpPromise.h	Sat Feb 26 08:40:25 2022 +0000
+++ b/dom/serviceworkers/ServiceWorkerOpPromise.h	Sat Feb 26 08:40:26 2022 +0000
@@ -30,10 +30,14 @@
 using FetchEventRespondWithPromise =
     MozPromise<FetchEventRespondWithResult, CancelInterceptionArgs, true>;
 
+using FetchEventPreloadResponseArgs =
+    Tuple<SafeRefPtr<InternalResponse>, IPCPerformanceTimingData, nsString,
+          nsString>;
+
 // The reject type int is arbitrary, since this promise will never get rejected.
 // Unfortunately void is not supported as a reject type.
 using FetchEventPreloadResponsePromise =
-    MozPromise<SafeRefPtr<InternalResponse>, int, true>;
+    MozPromise<FetchEventPreloadResponseArgs, int, true>;
 
 using ServiceWorkerOpPromise =
     MozPromise<ServiceWorkerOpResult, nsresult, true>;
